doc/libspectrum.3
.\" -*- nroff -*-
.\"
.\" libspectrum.3: libspectrum summary
.\" Copyright (c) 2004-2013 Philip Kendall
.\"
.\" This program is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2 of the License, or
.\" (at your option) any later version.
.\"
.\" This program is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License along
.\" with this program; if not, write to the Free Software Foundation, Inc.,
.\" 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
.\"
.\" Author contact information:
.\"
.\" E-mail: philip-fuse@shadowmagic.org.uk
.\"
.\"
.TH libspectrum 3 "xxth May, 2013" "Version 1.1.1" "Emulators"
.\"
.\"------------------------------------------------------------------
.\"
.SH NAME
libspectrum \(em Sinclair ZX Spectrum support library
.\"
.\"------------------------------------------------------------------
.\"
.SH DESCRIPTION
.I libspectrum
is a library for reading and writing some of the file formats used by
emulators of the Sinclair ZX Spectrum. It also contains routines
providing information about the various Spectrum clones which were
available.
.PP
The file formats currently supported by
.I libspectrum
are:
.IP \(bu 2
Snapshots: .z80, .szx, .sna (all read/write), .zxs, .sp., .snp and +D
snapshots (read only).
.IP \(bu 2
Tape images: .tzx, .tap, .spc, .sta, .ltp (read/write) and .pzx, Warajevo .tap,
Z80Em and CSW version 1 (read only).
.IP \(bu 2
Input recordings: .rzx (read/write).
.IP \(bu 2
Disk images: .dsk (both plain and extended), .fdi, .sad, .scl, .td0, .trd
and .udi (identification only).
.IP \(bu 2
Timex cartridges: .dck (read only).
.IP \(bu 2
IDE hard disk images: .hdf (read/write).
.IP \(bu 2
Microdrive cartridge images: .mdr (read/write).
.PP
Additionally, any files compressed with
.IR bzip2 "(3)"
or
.IR gzip "(3)"
can be read and will be transparently decompressed.
.PP
Full documentation for
.I libspectrum
is provided in
.RI ` libspectrum.txt '.
.\"
.\"------------------------------------------------------------------
.\"
.SH SEE ALSO
.IR bzip2 "(3),"
.IR fuse "(1),"
.IR fuse\-utils "(1),"
.IR gzip "(3)"
.PP
The comp.sys.sinclair Spectrum FAQ, at
.br
.IR "http://www.worldofspectrum.org/faq/index.html" .
.\"
.\"------------------------------------------------------------------
.\"
.SH AUTHOR
Philip Kendall (philip\-fuse@shadowmagic.org.uk)


ChangeLog
2013-05-xx  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 1.1.1 released.

        * Decreased libtool version to allow binaries built against
          libspectrum 1.0.0 to run with libspectrum 1.1.1.

2013-05-18  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 1.1.0 released.

        * Support for Spectranet in snapshots (Philip Kendall).

        * Support for Cheetah SpecDrum in snapshots (Fredrick Meunier).

        * Support reading PZX files (Fredrick Meunier).

        * Support for the TZX set signal level block (Fredrick Meunier).

        * Various minor bug fixes/improvements:
          * Add support for detecting POK files (Sergio Baldoví).
          * Advertise support for bzip2 compressed and wav files (Sergio
            Baldoví).
          * Drop support for GLib version 1.x (Alberto Garcia).
          * Add support for audiofile 0.3.x (Alberto Garcia).
          * Load all snapshot RAM pages for machines with more than 128k 
            (Fredrick Meunier; thanks, ketmar).
          * Support more SZX 1.5 features (Fredrick Meunier).
          * Modernise autoconf support (Sergio Baldoví).

2010-12-16  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 1.0.0 released.

        * New end of tape flag (Fredrick Meunier).

        * AF and AF' were swapped when reading and writing SZX files
          (Fredrick Meunier; thanks, velesoft).

        * Support for Fuller Box and Melodik (and similar AY interfaces) in
          snapshots (Fredrick Meunier).

        * Support for Opus Discovery in snapshots (Fredrick Meunier, Gergely
          Szasz).

        * Support for the NTSC 48K Spectrum (Philip Kendall).

        * Support for stereo sound files (Fredrick Meunier; thanks, Gilberto
          Almeida).

        * Add proper support for +2A/+3, Pentagon and Scorpion in .z80 files
          (Philip Kendall; thanks, Nick Corey)

        * Various fixes for compiling with Visual Studio 6 (Simon Owen).

        * Various minor bug fixes/improvements:
          * Looser signature checks for .dsk files (Gergely Szasz; thanks,
            zx81 and Simon Owen).
          * --without-(bzip2|zlib) options to configure for Gentoo and similar
            distributions (José Manuel Ferrer Ortiz).
          * Handle querying the block length for turbo and pure data blocks
            with 0 length (Fredrick Meunier).
          * Fix loading of Warajevo TAPs with raw samples (Fredrick Meunier).
          * Fail sensibly if file decompression fails (Fredrick Meunier).
          * Remove various unused variables, initialisations etc (Fredrick
            Meunier).

2008-11-30  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 0.5.0 released.

	* Memory management change: memory allocation functions now abort
	  on failure. Leads to simpler code both internally and for
	  users of the library (Philip Kendall).

	* Support for tape acceleration (Philip Kendall)

	* IDE improvements to allow DivIDE to support FATware (Philip
	  Kendall; thanks, http://www.dusky.sk/zxs/)

	* New API to deal with flagging automatic snapshots in RZX files
	  (Philip Kendall).

	* New function to get length (time) of a tape block (Philip Kendall,
	  Fredrick Meunier).

	* 128K SNA files are now treated as being for the Pentagon 128
	  rather than the Spectrum 128 (Fredrick Meunier)

	* SNA files now written out correctly (Philip Kendall).

	* Handle MDR write protection flag correctly (Philip Kendall)

	* Allow MDR images with bad checksums to load (the error will
	  be handled by the If1 ROM) (Gergely Szasz)

        * Support saving custom ROMs in SZX snapshots (Fredrick Meunier,
	  Stuart Brady)

	* Support for DivIDE, Kempston mouse and the simple IDE interface
	  in SZX snapshots (Fredrick Meunier)

        * Allow building without libgcrypt (Stuart Brady)

	* Amiga and MorphOS compilation fixes (Chris Young, Q-Master)

	* Don't produce empty libraries to fix build process on
	  OpenSolaris (Fredrick Meunier)

	* Remove deprecated functions (Fredrick Meunier).

	* Tweak timings for TS2068 (Fredrick Meunier; thanks Philip Kendall).

        * NTSC Spectrum support (Philip Kendall; thanks, Claudio Bernet).

2008-01-05  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

	* libspectrum 0.4.0 released.

	* Add support for loading from WAV files (Fredrick Meunier).

	* Add support for CSW tape files (Fredrick Meunier).

	* Tweak timings for TC2048 and TC2068 (Fredrick Meunier; thanks Philip
	  Kendall and Mark Woodmass).

	* Add support for the +D in z80 and szx snapshots (Fredrick Meunier).

	* Add Pentagon 512/1024 support (Q-Master).

	* Allow Amiga version to decompress files with xfdmaster.library
	  (Chris Young).

	* Allow MorphOS version to use xfdmaster.library (Q-Master).

	* Various other bugfixes.

2007-05-11  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

	* libspectrum 0.3.0.1 released (bug fix release).

	* Avoid crash when reading an invalid TZX file (thanks, crabfists).
	  (Philip Kendall).

	* Export the TZX Generalized Data Block functions when building a
	  Win32 DLL (Stuart Brady).

	* Distribute the hacking/ directory (thanks, Stuart Brady) (Philip
	  Kendall).

	* Remove warning when building from CVS (Stuart Brady).

2007-04-11  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

	* libspectrum 0.3.0 released.

	* Support for RZX rollback, including having snapshots and
	  multiple input recording blocks in input recording files (Philip
	  Kendall).

	* Improved IDE emulation (Matthew Westcott).

	* Support for microdrive images (Philip Kendall, Gergely Szasz).

	* TZX Generalized Data Block support. (Philip Kendall).

	* Support for Z80Em and CSW tape files (Darren Salt).

	* Support for the TS2068 (Fredrick Meunier).

	* Support for connected joystick information in snapshots (Fredrick
	  Meunier).

	* Add interrupt length to timing information (Philip Kendall).


website/libspectrum.php
<?php

/* libspectrum.php: the libspectrum 'home' page
   Copyright (c) 1999-2008 Darren Salt, Philip Kendall

   $Id$

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License along
   with this program; if not, write to the Free Software Foundation, Inc.,
   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

   Author contact information:

   E-mail: philip-fuse@shadowmagic.org.uk

*/

include ".inc/fuse.inc";

fuse_title ("libspectrum - emulator support library");
fuse_menu_heading ("libspectrum");

#<!-- ======================================================= -->

fuse_section (NULL, "What is it?", <<<END_SECTION
  <p>libspectrum is a library designed to make the input and output of some ZX Spectrum emulator files slightly easier. It is intended to be usable on Unix variants, Win32 and Mac OS X. Currently, it is mainly (only?) used by <a href="fuse.php">Fuse</a>, but other Spectrum emulator and utility authors are encouraged to use it.</p>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("Supported formats", "Which formats does it support?", <<<END_SECTION
  <ul>
   <li>Snapshots:
    <ul>
     <li>read/write: .z80, .szx, .sna</li>
     <li>read only: .zxs, .sp., .snp, +D snapshots</li>
    </ul>
   </li>
   <li>Tape images:
    <ul>
     <li>read/write: .tzx, .tap, .spc, .sta, .ltp</li>
     <li>read only: Warajevo .tap, Z80Em, CSW version 1, .wav</li>
    </ul>
   </li>
   <li>Input recordings:
    <ul><li>read/write: .rzx</li></ul>
   </li>
   <li>Timex cartridges:
    <ul><li>read only: .dck</li></ul>
   </li>
   <li>IDE hard disk images:
    <ul><li>read/write: .hdf</li></ul>
   </li>
   <li>Microdrive cartridge images:
    <ul><li>read/write: .mdr</li></ul>
   </li>
   <li>Compressed files: reading of gzipped and bzipped files is transparently supported.</li>
  </ul>

  <p>There are also some useful utility routines to do things like identify Spectrum files, or give the timings for various machines. For more details, see the <a href="libspectrum.txt">supplied documentation</a>.</p>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("Download", "Downloads", <<<END_SECTION

  <p>libspectrum is licensed under the <a href="COPYING">GNU General Public License</a>. Please read this before downloading libspectrum if you're not already familiar with it.</p>

  <h3>Source</h3>
  <ul>
   <li><a href="http://prdownloads.sourceforge.net/fuse-emulator/libspectrum-0.5.0.1.tar.gz?download">libspectrum-0.5.0.1.tar.gz</a> (<a href="http://prdownloads.sourceforge.net/fuse-emulator/libspectrum-0.5.0.1.tar.gz.sig?download">PGP signature</a>) or from <a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/libspectrum-0.5.0.1.tar.gz">World of Spectrum</a> (<a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/libspectrum-0.5.0.1.tar.gz.sig">signature</a>).</li>
   <li>The source code release above is signed with the <a href="http://wwwkeys.pgp.net:11371/pks/lookup?op=get&amp;search=0xD0767AB9">Fuse Release Key, ID <tt>D0767AB9</tt></a>, which has fingerprint <tt>064E 0BA9 688F 2699 3267 B1E5 1043 EEEB D076 7AB9</tt>. This is different from the key used to sign the 0.1.1 release as I forgot the passphrase for that key <tt>:-(</tt>.</li>
  </ul>

  <h3>Binaries</h3>
  <p>Packages are available for some Unix distributions; in general, any problems which are specific to the packages should be sent to the package maintainer.</p>
  <ul>
   <li>Unofficial packages of 0.5.0.1 for <a href="http://www.debian.org/">Debian</a> (or <a href="http://www.ubuntu.com/">Ubuntu</a>) by Alberto Garcia are available from <a href="http://people.igalia.com/berto/">his personal webpage</a> (i386 binaries and source).</li>
   <li><a href="https://admin.fedoraproject.org/pkgdb/packages/name/libspectrum">Fedora</a> (0.5.0)</li>
   <li><a href="http://www.FreeBSD.org/cgi/cvsweb.cgi/ports/emulators/libspectrum/">FreeBSD</a> (0.5.0)</li>
   <li><a href="http://www.netbsd.org/">NetBSD</a> has 0.5.0.1 available as <a href="http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/emulators/libspectrum/">emulators/libspectrum</a> in <a href="http://www.pkgsrc.org/">pkgsrc</a>.</li>
   <li><a href="http://www.openbsd.org/">OpenBSD</a> users can find a port of 0.5.0 in <a href="http://www.openbsd.org/cgi-bin/cvsweb/ports/devel/libspectrum/">ports/devel/libspectrum</a>.</li>
   <li><a href="http://packages.gentoo.org/package/app-emulation/libspectrum?full_cat">Gentoo</a> (0.4.0)</li>
   <li><a href="http://www.mandriva.com/">Mandriva</a> packages of 0.4.0 are available from the <a href="http://plf.zarb.org/">PLF</a>, or some <a href="http://www.speccy.org/sinclairmania/arch/emu/mandrake/">older packages</a> from Miguel Barrio Orsikowsky.</li>
   <li><a href="libspectrum-0.4.0-1.zj.i586.rpm">0.4.0</a> for <a href="http://www.suse.com">SuSE</a> 9.3, with thanks to Zoran Jekic.</li>
   <li><a href="http://en.docs.pld-linux.org/">PLD</a> users have a package of 0.2.1 available.</li>
   <li><a href="mailto:spec(at)webtech(dot)pl">Marek Januszewski</a> has made a <a href="libspectrum-0.2.0.1-i386-1spec.tgz">Slackware 9.1</a> (<a href="libspectrum-0.2.0.1-i386-1spec.tgz.md5">md5 sum</a>) package of 0.2.0.1.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section (NULL, "Development", <<<END_SECTION
  <p>If you're interested in working on libspectrum, subscribe to the <a href="http://lists.sourceforge.net/lists/listinfo/fuse-emulator-devel">fuse-emulator-devel</a> mailing list and/or see the <a href="http://sourceforge.net/projects/fuse-emulator/">project page</a> on SourceForge. If you just want news of new versions, you probably want the (low volume) <a href="http://lists.sourceforge.net/lists/listinfo/fuse-emulator-announce">fuse-emulator-announce</a> list instead.</p>

  <p>Finally, the latest development version should always be available as the 'trunk/libspectrum' directory from the <a href="http://sourceforge.net/svn/?group_id=91293">Subversion repository</a> on SourceForge. Don't expect any support for this, or even that it can be compiled...</p>

END_SECTION
);

#<!-- ======================================================= -->

fuse_related ("Fuse", "fuse.php");

fuse_footer ();
?>
